From 1f45f8a3f49e778aed940326e9abf0ea89b8aa1c Mon Sep 17 00:00:00 2001
From: Bas Zalmstra <zalmstra.bas@gmail.com>
Date: Tue, 1 Oct 2024 15:01:26 +0200
Subject: [PATCH] use gnu install dirs

---
 Build/CMakeLists.txt | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/Build/CMakeLists.txt b/Build/CMakeLists.txt
index bf40d940a..922dd68f1 100644
--- a/Build/CMakeLists.txt
+++ b/Build/CMakeLists.txt
@@ -282,14 +282,18 @@ endif()
 
 # Install Jolt library and includes
 if (ENABLE_INSTALL)
+	include(GNUInstallDirs)
 	install(TARGETS Jolt
 		EXPORT JoltExport
-		DESTINATION lib)
+		INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+		LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+		ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+		RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
 	foreach(SRC_FILE ${JOLT_PHYSICS_SRC_FILES})
 		string(REPLACE ${PHYSICS_REPO_ROOT} "" RELATIVE_SRC_FILE ${SRC_FILE})
 		get_filename_component(DESTINATION_PATH ${RELATIVE_SRC_FILE} DIRECTORY)
 		if (NOT RELATIVE_SRC_FILE MATCHES "\.cpp")
-			install(FILES ${SRC_FILE} DESTINATION include/${DESTINATION_PATH})
+			install(FILES ${SRC_FILE} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${DESTINATION_PATH})
 		endif()
 	endforeach()
 
@@ -298,7 +302,7 @@ if (ENABLE_INSTALL)
 		NAMESPACE Jolt::
 		FILE JoltConfig.cmake)
 	install(EXPORT JoltExport
-		DESTINATION lib/cmake/Jolt/
+		DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/Jolt/
 		NAMESPACE Jolt::
 		FILE JoltConfig.cmake)
 endif()
